src/System.Reflection.Emit.ILGeneration/tests/Label/LabelEquals.cs

```diff
@@ -49,5 +49,14 @@ public void Equals_EqualityOperators()
             Assert.True(lb1 == lb2);
             Assert.False(lb1 != lb2);
         }
+
+        [Fact]
+        public void Equals_IEquatable()
+        {
+            IEquatable<Label> label1 = new Label();
+            Label label2 = new Label();
+
+            Assert.True(label1.Equals(label2));
+        }
```

Comment:
Nit: this test would pass if the implementation of `IEquatable<Label>.Equals` were just `=> true`.  Could we fill it out a bit more?  We could also just add relevant lines to existing test cases above instead of adding this test, e.g. add:
```C#
Assert.True(((IEquatable<Label>)label).Equals((IEquatable<Label>)label));
```
to the `Equals_SameInstance_ReturnsTrue` test, and adding:
```C#
Assert.False(((IEquatable<Label>)label1).Equals((IEquatable<Label>)label2));
```
to the `Equals_DifferentLabel_ReturnsFalse` test.

Reply:
That's a fair point. I'll do some tinkering, though I do like your idea of merging with the already existing equality tests.

Reply:
I've made the recommended changes, though I've removed the redundant casts.